src/adv.py

```diff
@@ -19,33 +22,98 @@
 
 # Write a loop that:
-#
-# * Prints the current room name
-# * Prints the current description (the textwrap module might be useful here).
-# * Waits for user input and decides what to do.
-#
-# If the user enters a cardinal direction, attempt to move to the room there.
-# Print an error message if the movement isn't allowed.
-#
-# If the user enters "q", quit the game.
+while True:
+    # * Prints the current room name
+    # * Prints the current description (the textwrap module might be useful here).
+    current_room = new_player.current_room
+    print(f"Current room: {current_room.name}")
+    print(f"Description: {current_room.description}")
+    
+    # * Waits for user input and decides what to do.
+    print("---------------------------")
+    print("       Which direction would you like to go?")
+    direction = input("[n] North  [s] South   [e] East   [w] West  [q] Quit\n>>>")
+
+    # If the user enters a cardinal direction, attempt to move to the room there.
+    # Print an error message if the movement isn't allowed.
+    if direction == "n":
+        print("You selected North.".upper())
+        if current_room.n_to is not None:
+            new_player.current_room = current_room.n_to
+            print(f"You are now entering the {new_player.current_room.name}!!")
+            print("---------------------------")
+        else:
+            print(":( Im sorry, you've reached a dead end...")  
+            print("---------------------------")
+    elif direction == "s":
+        print("You selected South.".upper())
+        if current_room.s_to is not None:
+            new_player.current_room = current_room.s_to
+            print(f"You are now entering the {new_player.current_room.name}!!")
+            print("---------------------------")
+        else:
+            print(":( Im sorry, you've reached a dead end...") 
+            print("---------------------------")
+    elif direction == "e":
+        print("You selected East.".upper())
+        if current_room.e_to is not None:
+            new_player.current_room = current_room.e_to
+            print(f"You are now entering the {new_player.current_room.name}!!")
+            print("---------------------------")
+        else:
+            print(":( Im sorry, you've reached a dead end...") 
+            print("---------------------------")
+    elif direction == "w":
+        print("You seleected West.".upper())
+        if current_room.w_to is not None:
+            new_player.current_room = current_room.w_to
+            print(f"You are now entering the {new_player.current_room.name}!!")
+            print("---------------------------")
+        else:
+            print(":( Im sorry, you've reached a dead end...".upper()) 
+            print("---------------------------")
+    elif direction == "q":
+        # If the user enters "q", quit the game.
+        print("Ciao Bella! o.0")
+        break
+    else:
+        print("Not a valid input")
```

Comment:
Try to separate movement from the main game loop. Who is moving? where should move function belong?